src/main/java/com/mixpanel/android/mpmetrics/DecideMessages.java

```diff
@@ -135,96 +48,12 @@ public synchronized void reportResults(List<InAppNotification> newNotifications,
                     mIntegrations = integrationsSet;
-                    newContent = true;
+                    mConnectIntegrations.setupIntegrations(mIntegrations);
                 }
             } catch(JSONException e) {
                 MPLog.e(LOGTAG, "Got an integration id from " + integrations.toString() + " that wasn't an int", e);
             }
         }
-
-        MPLog.v(LOGTAG, "New Decide content has become available. " +
-                    newNotifications.size() + " notifications and " +
-                    variants.length() + " experiments have been added.");
-
-        if (newContent && null != mListener) {
-            mListener.onNewResults();
-        }
-    }
-
-    public synchronized JSONArray getVariants() {
-        return mVariants;
-    }
-
-    public synchronized InAppNotification getNotification(boolean replace) {
-        if (mUnseenNotifications.isEmpty()) {
-            MPLog.v(LOGTAG, "No unseen notifications exist, none will be returned.");
-            return null;
-        }
-        InAppNotification n = mUnseenNotifications.remove(0);
-        if (replace) {
-            mUnseenNotifications.add(n);
-        } else {
-            MPLog.v(LOGTAG, "Recording notification " + n + " as seen.");
-        }
-        return n;
-    }
-
-    public synchronized InAppNotification getNotification(int id, boolean replace) {
-        InAppNotification notif = null;
-        for (int i = 0; i < mUnseenNotifications.size(); i++) {
-            if (mUnseenNotifications.get(i).getId() == id) {
-                notif = mUnseenNotifications.get(i);
-                if (!replace) {
-                    mUnseenNotifications.remove(i);
-                }
-                break;
-            }
-        }
-        return notif;
-    }
-
-    public synchronized InAppNotification getNotification(AnalyticsMessages.EventDescription eventDescription, boolean replace) {
-        if (mUnseenEventTriggeredNotifications.isEmpty()) {
-            MPLog.v(LOGTAG, "No unseen triggered notifications exist, none will be returned.");
-            return null;
-        }
-        for (int i = 0; i < mUnseenEventTriggeredNotifications.size(); i ++) {
-            final InAppNotification n = mUnseenEventTriggeredNotifications.get(i);
-            if (n.matchesEventDescription(eventDescription)) {
-                if (!replace) {
-                    mUnseenEventTriggeredNotifications.remove(i);
-                    MPLog.v(LOGTAG, "recording triggered notification " + n.getId() +
-                            " as seen " + eventDescription.getEventName());
-                }
-                return n;
-            } else {
-                MPLog.v(LOGTAG, "triggered notification " + n.getId() +
-                        " does not match event " + eventDescription.getEventName());
-            }
-        }
-        return null;
-    }
-
-    public synchronized Set<String> getIntegrations() { return mIntegrations; }
```

Comment:
is this no longer needed?

Reply:
we use `mIntegrations` directly within the class instead, so no public method is needed.